lib/board.rb

```diff
@@ -0,0 +1,71 @@
+class Board
+  attr_accessor :cells, :token_1, :token_2
+
+  def initialize(token_1 = 'X', token_2 = 'O')
+    self.reset!
+    self.token_1 = token_1
+    self.token_2 = token_2
+  end
+
+  def reset!
+    self.cells = Array.new(9, ' ')
+  end
+
+  def display
+    puts board_row(0)
+    puts horizontal_line
+    puts board_row(1)
+    puts horizontal_line
+    puts board_row(2)
+  end
+
+  def board_row(index)
+    board = self.cells
+    first = board[0 + index * 3]
+    second = board[1 + index * 3]
+    third = board[2 + index * 3]
+    ' ' + first + ' | ' + second + ' | ' + third + ' '
+  end
+
+  def horizontal_line
+    puts '-----------'
+  end
+
+  def position(input)
+    self.cells[input.to_i - 1]
+  end
+
+  def full?
+    self.turn_count == 9
+  end
+
+  def taken?(input)
+    position(input) != ' '
+  end
+
+  def turn_count
+    turn_count = 0
+    ('1'..'9').each do |index|
+      turn_count += 1 if taken?(index)
+    end
+    turn_count
+  end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  (1..9).count{|index| taken?(index)}
end
```